File: practice/7_Reverse_Integer.js
function reverse(x) {
    let reversed = 0;

    while (x !== 0) {
        const digit = x % 10;
        x = Math.trunc(x / 10);

        if (
            reversed > Math.pow(2, 31) / 10 ||
            reversed < Math.pow(-2, 31) / 10
        ) {
            return 0;
        }

        reversed = reversed * 10 + digit;
    }

    return reversed;
}

console.log(reverse(321));
console.log(reverse(-345));
console.log(reverse(3210));

File: practice/7_Reverse_Integer.kt
fun main() {
	println(revInt(321));
println(revInt(-345));
println(revInt(3210));
}

fun revInt(x: Int): Int {
	var num = x
	var reversed = 0

	while(x != 0) {
		val digit = num%10
		num=x/10

		reversed = reversed * 10 + digit
	}

	return reversed
}
